### PR TITLE
Add wavelength to PeakCollection

### LINKED ISSUE #318: PeakCollection work with wavelength
There should be an additional attribute on `PeakCollection` for wavelength. This will be used for calculating the d-spacing of a peak. As a result the accessor methods will be:
* `get_native_parameters()` to get the parameters of the peak function
* `get_effective_parameters()` to get the center, height, fwhm, mixing, A0, A1 and intensity
* `get_dspacing_center()` to get the effective center and uncertainty converted to d-spacing

Second part is to modify `pyrs.core.summary_generator.SummaryGenerator` with the `<peak_tag>_dspacing_center` being the first column in the peak section and `<peak_tag>_dspacing_center_error` being the first column in the peak uncertainty section.

### PR DESCRIPTION
**Pull request by Fahima-Islam**
added "dspacing center"

Fixes #318

### CODE REVIEW

File: pyrs/peaks/peak_collection.py [hunk 1]
```diff
@@ -180,7 +180,8 @@ def get_dspacing_center(self):
             A two-item tuple containing the peak center and its uncertainty.
         """
         effective_values, effective_errors = self.get_effective_params()
-        theta_center_value, theta_center_error = effective_values['Center'] / 2, effective_errors['Center'] / 2
+        theta_center_value = np.array([value['Center'] / 2 for value in effective_values])
```
Comment (peterfpeterson) [inline pyrs/peaks/peak_collection.py]: Use implicit looping with numpy. Also, multiply is (generally) faster than divide
```suggestion
        theta_center_value = 0.5 * value['Center']
```

File: pyrs/peaks/peak_collection.py [hunk 2]
```diff
@@ -180,7 +180,8 @@ def get_dspacing_center(self):
...
+        theta_center_error = np.array([error['Center'] / 2 for error in effective_errors])
```
Comment (peterfpeterson) [inline pyrs/peaks/peak_collection.py]:
```suggestion
        theta_center_error = 0.5 * error['Center']
```

File: tests/unit/test_peak_collection.py [hunk 3]
```diff
@@ -5,13 +5,34 @@
 import pytest
 
 
-def test_peak_shape_enum():
-    assert PeakShape.getShape('gaussian') == PeakShape.GAUSSIAN
-    assert PeakShape.getShape('GAUSSIAN') == PeakShape.GAUSSIAN
+def check_peak_shape_enum(peak_shape, num_native_params):
```
Comment (peterfpeterson) [inline tests/unit/test_peak_collection.py]: The parameterization of the test is a good idea. Lines 8-25 could be converted into `@pytest.mark.parametrize` and possibly be clearer. I don't think it is necessary though unless we expand to test more peak shapes.

File: tests/unit/test_peak_collection.py [hunk 4]
```diff
@@ -53,13 +103,45 @@ def test_peak_collection():
...
+
+
+def test_peak_collection_Gaussian():
+    NUM_SUBRUN = 2
+    # without wavelength
+    check_peak_collection('Gaussian', NUM_SUBRUN, np.zeros(NUM_SUBRUN, dtype=get_parameter_dtype(effective=True)))
+    # with wavelength
+    check_peak_collection('Gaussian', NUM_SUBRUN, np.zeros(NUM_SUBRUN, dtype=get_parameter_dtype(effective=True)),
+                          wavelength=3.15, target_d_spacing_center=[1.57, 1.73],
+                          target_d_spacing_center_error=[0.0, 0.0])
```
Comment (Fahima-Islam) [inline tests/unit/test_peak_collection.py]: @peterfpeterson line 126 actually is the test without setting wavelength that would confirm  that the d-spacing returned is nan

**Comment by Fahima-Islam** [comment]
> Mostly minor comments. Please add an additional test to `test_peak_collection.py` without setting wavelength to confirm that the d-spacing returned is `nan`.

@peterfpeterson line 126 and 136 are actually  the test without setting wavelength that would confirm that the d-spacing returned is nan as the default value of target_d_spacing_center and target_d_spacing_center_error are np.nan.

File: tests/unit/test_peak_collection.py (hunk 4 above)
Comment (peterfpeterson) [reply tests/unit/test_peak_collection.py]: I completely missed the comment pointing that out :man_facepalming:

**Pull request merged**